djangoProject3/DFS/models.py

```diff
@@ -0,0 +1,107 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from rest_framework.authtoken.models import Token
+
+
+# Create your models here.
+# This signal creates auth token when user is created
+@receiver(post_save, sender=settings.AUTH_USER_MODEL)
+def create_auth_token(sender, instance=None, created=False, **kwargs):
+    if created:
+        Token.objects.create(user=instance)
+
+
+class UserProfile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    profile_picture = models.ImageField(upload_to='profilePics/', null=True, blank=True)
+    bio = models.TextField(max_length=500, blank=True)
+
+    def __str__(self):
+        return self.user.username
+
+
+class Author(models.Model):
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+class Genre(models.Model):
+    name = models.CharField(max_length=50)
+
+    def __str__(self):
+        return self.name
+
+
+class Book(models.Model):
+    title = models.CharField(max_length=200)
+    author = models.ForeignKey(Author, on_delete=models.CASCADE)
+    genres = models.ManyToManyField(Genre)
+    price = models.DecimalField(max_digits=10, decimal_places=2)
+    price_id = models.CharField(max_length=50)
+
+    def __str__(self):
+        return self.title
+
+
+class Review(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    book = models.ForeignKey(Book, on_delete=models.CASCADE)
+    text = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Rating(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    book = models.ForeignKey(Book, on_delete=models.CASCADE)
+    VALUE_CHOICES = (
+        (1, '1'),
+        (2, '2'),
+        (3, '3'),
+        (4, '4'),
+        (5, '5'),
+    )
+    rating = models.IntegerField(choices=VALUE_CHOICES)
+
+
+class Notification(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    event_type = models.CharField(max_length=100)
+    message = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    is_read = models.BooleanField(default=False)
+
+
+class Order(models.Model):
+    COMPLETED_CHOICES = [
+        ('unprocessed', 'Unprocessed'),
+        ('processing', 'Processing'),
+        ('on the way', 'On the Way'),
+        ('delivered', 'Delivered'),
+    ]
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    total_price = models.DecimalField(max_digits=10, decimal_places=2)
+    is_completed = models.CharField(
```

Comment:
this field name should be status, ```is_completed``` is giving the sense of a boolean flag not a dropdown of choices

Reply:
It contains option like delivered, processing and unprocessing

Reply:
field name should be set accordinlgy then to show that it is a choice field not a boolean flag